File: python/matrix/matrix.py
class Matrix:
    def __init__(self, matrix_string):
        self.matrix_string = self.make_matrix(matrix_string)

    def make_matrix(self, matrix_string):
        return [[int(item) for item in array.split()] for array in matrix_string.splitlines()]
        
    def row(self, index):
        return [item for item in self.matrix_string[index - 1]]

    def column(self, index):
        return [item[index - 1] for item in self.matrix_string]

File: JavaScript/matrix/matrix.js
//
// This is only a SKELETON file for the 'Matrix' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Matrix {
  constructor(matrixString = "") {
    this.matrix = []
    this.matrixColumns = []
    this.parseMatrix(matrixString)
  }

  parseMatrix(matrixString){
    matrixString.split("\n").forEach(array => this.parseArray(array)) 
  }
  parseArray(arrayString){
    let numArray = arrayString.split(" ").map(x=> Number(x))
    this.matrix.push(numArray)
  }

  parseColumns(row){
    for(let index=0; index<row.length; index++){
      this.matrixColumns.length < index+1 && this.matrixColumns.push(new Array()) 
      this.matrixColumns[index].push(row[index])
    }
      
  }

  get rows() {
    return this.matrix
  }

  get columns() {
    this.matrix.forEach(row => this.parseColumns(row))
    return this.matrixColumns
  }


}
